R/bayesiannetworkanalysis.R

```diff
@@ -193,67 +202,146 @@ BayesianNetworkAnalysis <- function(jaspResults, dataset, options) {
 
   networks <- vector("list", length(dataset))
 
-  # For every dataset (given the splitting variable) estimate a network:
+  # For every dataset (given the splitting variable) estimate a network: KEEP THIS AS IS FOR NOW
   for (nw in seq_along(dataset)) {
-
+    #ADD CONDITIONAL STATEMENTS HERE FOR EACH OF THE THREE OPTIONS
     # When method = "gcgm" a vector with binary values is needed:
-    nonContVariables <- NULL
-    if (options[["estimator"]] == "gcgm") {
+    # if model is gcgm
+    if (options[["model"]] == "ggm") {
+
+      # Estimate mixed network
+      jaspBase::.setSeedJASP(options)
+      easybgmFit <- try(easybgm::easybgm(data       = as.data.frame(apply(dataset[[nw]], 2, as.numeric)),
+                                         type       = "continuous",
+                                         package    = "BDgraph" ,
+                                         iter       = as.numeric(options[["iter"]]),
+                                         save       = TRUE,
+                                         centrality = FALSE,
+                                         burnin     = as.numeric(options[["burnin"]]),
+                                         g.start    = options[["initialConfiguration"]],
+                                         df.prior   = as.numeric(options[["dfprior"]]),
+                                         g.prior    = as.numeric(options[["gprior"]])))
```

Comment:
```suggestion
      easybgmFit <- try(easybgm::easybgm(data       = dataset[[nw]],
                                         type       = "continuous",
                                         package    = "BDgraph" ,
                                         iter       = options[["iter"]],
                                         save       = TRUE,
                                         centrality = FALSE,
                                         burnin     = options[["burnin"]],
                                         g.start    = options[["initialConfiguration"]],
                                         df.prior   = options[["dfprior"]],
                                         g.prior    = options[["gprior"]]))
```
so in principle, the data should already be in the right format at this point (e.g., a `data.frame` with only numeric values). In addition, the values from the options should also be actual values already. So the removed lines *should be* redundant. If these lines are somehow not redundant, then this should probably be fixed when reading the data.

Reply:
> I fixed most of the arguments. Except for transforming the variables to numeric when model == "ggm" || "gcgm". This is because if you pass non-numeric variables to easybgm when using BDgraph, an error occurs internally in the get_S_n_p function. I did not want to alter the .networkAnalysisReadData function, which is used for all network analyses. However, I am open to suggestions if you think this needs to be handled differently.

I think I found a solution for this!